website/content/docs/integrations/vault/acl.mdx

```diff
@@ -751,18 +751,26 @@ $ VAULT_TOKEN=s.H39hfS7eHSbb1GpkdzOQLTmz.fvuLy nomad job run vault.nomad
 ## Migrating to Using Workload Identity with Vault
 
 Migrating from the legacy (pre-1.7) workflow where workloads use the agent's
-Vault token requires configuation on your Vault cluster and your Nomad server
+Vault token requires configuration on your Vault cluster and your Nomad server
 agents. It does not require updating your running Nomad jobs unless you wish to
 specify a non-default role. To migrate:
 
 * Create the Vault auth method, default role, and policies on your Vault
   cluster.
 * Enable [`vault.default_identity`][] blocks in your Nomad server agent
-  configurations.
-* (Optionally) Add [`vault.role`][] fields to any Nomad jobs that will not use
-  the default role.
-* (Optionally) add [`identity`][] blocks to your jobs if you want to use a
-  different identity because of how your auth method and roles are configured.
+  configurations, but **do not modify any of the existing Vault
+  configuration**.
+* Upgrade your cluster following the documented [Upgrade
+  Process][docs_upgrade].
+* Resubmit Nomad jobs that need access to Vault to redeploy them with a new
+  workload identity for Vault.
```

Comment:
I was trying to think of ways we could help with this. Two ideas I had were:

1. A command that looks for the following artifacts and tell users if their clusters are ready for the migration.
  * jobs with `vault` blocks (and maybe `templates` with Vault-related functions? A `vault` block is not always required to access Vault) but no Vault identities.
  * unused `VaultAccessors` in state store (and potentially clean them without needing a leadership transition)
2. Some kind of metric that operators can monitor and wait until it reaches zero. But I'm not sure yet exactly what to measure.

Reply:
I like the idea of a command... maybe we could bake this into a `-check` flag for `nomad setup vault`? 

A metric would be nice, but the only reasonable place for it would be in the state store and we'd have to repopulate it during snapshot restore. Kind of messy.

Reply:
Cool! I will try to think a bit more where to place the command. I was thinking somewhere under `nomad operator` to indicate that this may require a management token since it needs to look all over the state store.

It could also be used for other upgrade checks in the future.